graal-js/src/com.oracle.truffle.js.test/src/com/oracle/truffle/js/test/interop/ESModuleTest.java

```diff
@@ -211,6 +215,356 @@ private static void deleteFiles(File[] filesArray) {
         }
     }
 
+    /**
+     * Test basic module block features via mimicking the {@code} testFunctionExport() +
+     */
+    @Test
+    public void testModuleBlockMimicFunctionExport() throws IOException {
+        File[] allFilesArray = null;
+
+        TestOutput out = new TestOutput();
+        try (Context context = JSTest.newContextBuilder().allowHostAccess(HostAccess.ALL).allowIO(true).err(out).out(out).option(
+                        JSContextOptions.CONSOLE_NAME, "true").option(JSContextOptions.UNHANDLED_REJECTIONS_NAME, "throw").build()) {
+            allFilesArray = prepareTestFileAndModules("resources/moduleBlock/moduleBlockFunctionExportTestMimic.js",
+                            "resources" + "/functionexportmodule.js");
+
+            Source mainSource = Source.newBuilder(ID, allFilesArray[0]).mimeType("application/javascript+module").build();
+
+            Value v = context.eval(mainSource);
+
+            System.out.println(out);
+
+            assertTrue(v.hasArrayElements());
+
+            System.out.println("Square test: " + v.getArrayElement(0));
+            System.out.println("Diagonal test: " + v.getArrayElement(1));
+            System.out.println("Sqrt test: " + v.getArrayElement(2));
+        } finally {
+            deleteFiles(allFilesArray);
+        }
+    }
+
+    /**
+     * Test module block prototype behavior
+     */
+    @Test
+    public void testModuleBlockPrototype() throws IOException {
+        File[] allFilesArray = null;
+
+        TestOutput out = new TestOutput();
+
+        try (Context context = JSTest.newContextBuilder().allowHostAccess(HostAccess.ALL).allowIO(true).err(out).out(out).option(
+                        JSContextOptions.CONSOLE_NAME, "true").option(JSContextOptions.UNHANDLED_REJECTIONS_NAME, "throw").build()) {
+            allFilesArray = prepareTestFileAndModules("resources/moduleBlock/moduleBlockPrototype.js",
+                            "resources" + "/functionexportmodule.js");
+
+            Source mainSource = Source.newBuilder(ID, allFilesArray[0]).mimeType("application/javascript+module").build();
+
+            Value v = context.eval(mainSource);
+
+            System.out.println(out);
+
+            assertTrue(v.hasArrayElements());
+
+            System.out.println("Square test: " + v.getArrayElement(0));
+            System.out.println("Diagonal test: " + v.getArrayElement(1));
+            System.out.println("Sqrt test: " + v.getArrayElement(2));
+        } finally {
+            deleteFiles(allFilesArray);
+        }
+    }
+
+    /**
+     * Test module block prototype behavior
+     */
+    @Test
+    public void testModuleBlockAsync() throws IOException {
```

Comment:
All the unit tests look very similar. Probably you can refactor them in a single method and pass in the `.js` input file as an argument

Reply:
Overlapping code was unified.